### ISSUE TITLE
Use of uninitialized value in subroutine entry at /usr/lib64/perl5/vendor_perl/Encode.pm line 174.

### ISSUE THREAD

**Issue by ppisar** [Test-More/Test2-Plugin-Cover#2]
Have you noticed that using Test2::Plugin::Cover triggers a warning in Encode module:

~~~~
$ perl -Iblib/{lib,arch} -e 'use Test2::Plugin::Cover; use Test2::V0 -target => q{Test2::Plugin::Cover};'
Use of uninitialized value in subroutine entry at /usr/lib64/perl5/vendor_perl/Encode.pm line 174.
# Seeded srand with seed '20210409' from local date.
# This test covered 0 source files.
# No tests run!
~~~~

This seems to appear in 0.000012 and does not depend on Encode or perl version.

**Comment by exodist** [comment]
I cannot reproduce this. I Tried with Test2-Plugin-Cover-0.000013  on perl 5.22.4 with threads, and v5.32.0 without threads. As well I tried messing around with the LANG and LC_ALL env variables as well to no effect.

We need to figure out what is set/different between my systems and yours to find the cause and potential fix for this.

**Comment by ppisar** [comment]
The first Test-Plugin-Cover commit which triggers the warning is:

commit 8df204b2949a8e94727a3ac3db1fdc58c073cbcf (HEAD, refs/bisect/bad)
Author: Chad Granum ***@***.***>
Date:   Thu Apr 8 09:28:05 2021 -0700

    Fix a segv

I was unable to find what else is required for the warning. I downgraded Encode
to a version found in 5.32.0, I tried LC_ALL=C, C.UTF-8, and en_US.UTF-8.
Without any effect. I use perl 5.32.1.

Here is a cpantesters result without the warning
<http://www.cpantesters.org/cpan/report/2e538c4e-99a4-11eb-9580-1e291f24ea8f>
a day after with the warning
<http://www.cpantesters.org/cpan/report/d160d24e-9980-11eb-b2a7-3d3b1f24ea8f>. Both Perl 5.32.0.

A difference shows:

@@ -51,7 +55,7 @@
     ------------------- -------- --------
     Carp                0        1.50    
     File::Spec          0        3.78    
-    Path::Tiny          0.048    0.114   
+    Path::Tiny          0.048    0.118   
     perl                5.012000 5.032000
     Test2::API          1.302166 1.302183
     Test2::EventFacet   1.302166 1.302183
@@ -60,7 +64,7 @@
 
     Module              Need     Have    
     ------------------- -------- --------
-    ExtUtils::MakeMaker 0        7.44    
+    ExtUtils::MakeMaker 0        7.60    
     Fcntl               0        1.13    
     Test2::V0           0.000130 0.000139
 
@@ -68,7 +72,7 @@
 
     Module              Need     Have    
     ------------------- -------- --------
-    ExtUtils::MakeMaker 0        7.44    
+    ExtUtils::MakeMaker 0        7.60    
 
 
 ------------------------------
@@ -105,26 +109,26 @@
 
     Module              Have    
     ------------------- --------
-    CPAN                2.27    
+    CPAN                2.28    
     CPAN::Meta          2.150010
     Cwd                 3.78    
-    ExtUtils::CBuilder  0.280234
-    ExtUtils::Command   7.44    
-    ExtUtils::Install   2.14    
-    ExtUtils::MakeMaker 7.44    
-    ExtUtils::Manifest  1.72    
+    ExtUtils::CBuilder  0.280236
+    ExtUtils::Command   7.60    
+    ExtUtils::Install   2.20    
+    ExtUtils::MakeMaker 7.60    
+    ExtUtils::Manifest  1.73    
     ExtUtils::ParseXS   3.40    
     File::Spec          3.78    
-    JSON                4.02    
-    JSON::PP            4.04    
+    JSON                4.03    
+    JSON::PP            4.06    
     Module::Build       0.4231  
-    Module::Signature   n/a     
+    Module::Signature   0.87    
     Parse::CPAN::Meta   2.150010
     Test::Harness       3.42    
     Test::More          1.302183
     YAML                1.30    
-    YAML::Syck          1.32    
-    version             0.9924  
+    YAML::Syck          1.34    
+    version             0.9928  
 
 
 --

**Comment by exodist** [comment]
Fixed in 0.000015  (well worked around, I commented out a section of borrowed XS I do not understand, problem went away but all the data I want to collect is still there, so it is fixed, but I am ignorant as to what was wrong or why the fix fixes it)
